### PR TITLE
Add diagrams for the different executor runtimes

### LINKED ISSUE #49915: Add documentation on how executors work
Executors are a blackbox. It takes time for devs and users to really understand _how_ executors work. Executors also have many different scenarios and function slightly differently based on the scenario.

We should add documentation, with diagrams, that explain how executors work.

## Done

- Update docs to include
    - The different supported scenarios/runtimes
        - `src` locally
        - Bianry
        - Docker
        - Firecracker
        - K8s (dind)
        - Native K8s
    - Description oh how executors work in the different scenarios
    - Include diagrams

### PR DESCRIPTION
**Pull request by Piszmog**
Closes #49915.

Added diagrams for each of the runtimes supported. Under each diagram, included a step-by-step of _what_ is going on. Went with PNG files as SVG files were severely messed up when exporting them (words overlapped, arrows went to nowhere, etc...).

Runtimes
- Local with `src-cli`
- Binary
- Docker
- Firecracker
- Kubernetes Native
- Kubernetes Docker-in-Docker

## TODO

- Comment out K8s Native section (will uncomment as we get closer to June release)

## Test plan

No tests, just doc changes.

### CODE REVIEW

File: doc/admin/executors/index.md [hunk 1]
```diff
@@ -45,13 +45,129 @@ Instead of performing this work within the Sourcegraph instance, where code is a
...
+3.  A user initiates a process that creates executor Jobs.
+4.  The executor instance "dequeues" a Job.
+5.  Executor calls the Sourcegraph API to clone a repository.
+    1.  The repositories are written to a directory.
+6.  A Docker Container is created for each "step."
+    1.  `src` is invoked for batch change Jobs
+    2.  Docker is invoked directly for other Jobs
+    3.  "Steps" are ran in sequential order.
+7.  Within each Docker container a single Firecracker VM is started
+    1.  The directory containing the repository is mounted to the VM.
```
Comment (efritz) [inline doc/admin/executors/index.md:102]: image

Don't we have one VM for the entire run, not each docker container? Also shouldn't the ordering of the boxes here be Firecracker (outside) then Docker (inside)?

Comment (Piszmog) [reply doc/admin/executors/index.md]: > The firecracker one is actually two layers of docker: ignite runs a docker container within which an ignite-spawn process runs a firecracker vm within which a docker container is run

Comment (Piszmog) [reply doc/admin/executors/index.md]: I'll fix the steps as the order is wrong for how steps are ran

**Review by eseliger — APPROVED** [review]
thanks for doing this!

side q: should we upload the diagrams to our assets bucket instead of this repo to keep the clone size small?

**Comment by eseliger** [comment]
this says "handbook" but should still work for the repo, too: https://handbook.sourcegraph.com/handbook/editing/handbook-images-video/#adding-images-to-google-cloud-storage :)

**Pull request merged**